src/services/anchor-service.ts

```diff
@@ -564,8 +564,36 @@ export default class AnchorService {
    * @private
    */
   static async _loadCandidate(candidate: Candidate, ceramicService: CeramicService): Promise<void> {
-    // Build multiquery
-    const queries = candidate.requests.map((request) => {
+    // First, load the current known stream state from the ceramic node
+    let stream
+    try {
+      stream = await ceramicService.loadStream(candidate.streamId)
+    } catch (err) {
+      logger.err(`Failed to load stream ${candidate.streamId.toString()}: ${err}`)
+      candidate.failAllRequests()
+      return
+    }
+
+    // Now filter out requests from the Candidate that are already present in the stream log
+    const missingRequests = candidate.requests.filter((req) => {
+      const found = stream.state.log.find(({ cid }) => {
+        return cid.toString() == req.cid
+      })
+      return !found
+    })
+
+    // If stream already knows about all CIDs that we have requests for, great!
+    if (missingRequests.length == 0) {
+      candidate.setTipToAnchor(stream)
+      return
+    }
+
+    // If there were CIDs that we have requests for but didn't show up in the stream state that
+    // we loaded from Ceramic, we can't tell if that is because those commits were rejected by
+    // Ceramic's conflict resolution, or if our local Ceramic node just never heard about those
+    // commits before.  So we build a multiquery including all missing commits and send that to
+    // Ceramic, forcing it to at least consider every CID that we have a request for.
+    const queries = missingRequests.map((request) => {
```

Comment:
Can we add a log here so that we can see if there are missing requests present? I suspect we want to see this info in the future.

Reply:
If the commit is still missing after the multiquery we log it.  You want to also log the commits that are missing from the initial load but might still show up after the multiquery?  Do you want each individual CID logged, or would just a count of how many are missing be enough?

Reply:
Yes. A number might be enough. What do you think?

Reply:
If the CIDs are included it would be easier to debug though.

Reply:
okay done